assets/js/script.js

```diff
@@ -0,0 +1,198 @@
+           grid[checker[i][0]][checker[i][1]] = 0
+           grid[checker[i][0] + 1][checker[i][1]] = 0
+           grid[checker[i][0] + 2][checker[i][1]] = 0
+         }
+       }
+       //reset checker
+       checker = []
+     }
+      replenishCells()
+    }
+
+
+   function checkForMatches() {
+     //checks if any matches exist on the whole board
+    gameChecker = []
+    for (var x = 0 ; x < matrix; x++) {
+      for (var y = 0; y < matrix; y++) {
+
+         //check for horizontal matches
+        if (grid[x][y] === grid[x][y+1] && grid[x][y] === grid[x][y+2]) {
+          gameChecker.push([x,y,'h'])
+        }
+        //check for vertical matches
+        if (grid[x+2]) {
+          if (grid[x][y] === grid[x+1][y] && grid[x][y] === grid[x+2][y]) {
+            gameChecker.push([x,y,'v'])
+          }
+        }
+      }
+    }
+    return gameChecker.length
+    }
+
+  function replenishCells() {
+    //replenish the empty cells where matches were found by the player
+    var pos = x +","+ y
+
+    //replenish from bottom
+    for (var x = matrix - 1 ; x >= 0 ; x--) {
+      for (var y = matrix - 1; y >= 0 ; y--) {
+        if (grid[x-1] && grid[x][y] === 0) {
+          ;[ grid[x][y],grid[x-1][y] ] = [grid[x-1][y], grid[x][y] ]
+          pos = x +","+ y
+          $('.cell[data-position= "'+ pos +'"]').css('backgroundColor', grid[x][y])
+          }
+      }
+    }
+    //run again to replenish adjacent empty
+    for (var x = 0 ; x < matrix ; x++) {
+      for (var y = 0; y < matrix ; y++) {
+        if (grid[x-1] && grid[x][y] === 0) {
+          ;[ grid[x][y],grid[x-1][y] ] = [grid[x-1][y], grid[x][y] ]
+          pos = x +","+ y
+          $('.cell[data-position= "'+ pos +'"]').css('backgroundColor', grid[x][y])
+          }
+
+      }
+    }
+    //replenish last adjacent empty
+    for (var x = 0 ; x < matrix ; x++) {
+      for (var y = 0; y < matrix ; y++) {
+        if (grid[x][y] === 0) {
+          grid[x][y] = generateColor()
+          pos = x +","+ y
+         $('.cell[data-position= "'+ pos +'"]').css('backgroundColor', grid[x][y])
+        }
+      }
+    }
+
+    if (checkForMatches() === 0 ) {
+      //if no matches remain on the board, start a new level
+      alert('Level Complete!')
+      restart()
+    }
+  }
+
+  function generateColor() {
+    return colorList[Math.floor(Math.random() * colorList.length)]
+  }
+
+  function restart() {
+    //restarts level
+    grid = []
+    generateLevel()
+  }
+
+  function resetAll() {
+    //resets the game & starts a new game
+    clearInterval(timer)
+    time = 60
+    score = 0
+    scoreboard()
+    timer = setInterval(timerFn, 1000)
+    restart()
+  }
+
+  function scoreboard() {
+    //display score
+    $('h3').text('Score: ' + score)
+  }
+
+  function bgChanger(){
+    //background color changer
+    var r = Math.floor(Math.random() * 255)
+    var g = Math.floor(Math.random() * 255)
+    var b = Math.floor(Math.random() * 255)
+    var $body = $('body')
```

Comment:
You may move line 173 to outside of the function. Then what will be the benefit?